src/tree/v2/ResourceTreeDataProviderBase.ts

```diff
@@ -65,7 +66,10 @@ export abstract class ResourceTreeDataProviderBase extends vscode.Disposable imp
 
     async getTreeItem(element: ResourceGroupsItem): Promise<vscode.TreeItem> {
         const treeItem = await element.getTreeItem();
-        treeItem.id = this.itemCache.getId(element);
+
+        if (!(element instanceof BranchDataProviderItem)) {
```

Comment:
Is there a way to avoid special casing not-`BranchDataProviderItem`?  Perhaps by ensuring that all such cases return a tree item with `id` set, so that the case reduces to just setting `id` if not already set?